### PR TITLE
Enforce test results on Plone >= 4.3

### PR DESCRIPTION
**Pull request by hvelarde**

### CODE REVIEW

File: src/collective/cover/tests/test_banner_tile.robot [hunk 1]
```diff
@@ -21,8 +24,8 @@ ${edit_link_selector}  a.edit-tile-link
 *** Test cases ***
 
 Test Banner Tile
-    # XXX: test is randomly failing under Plone 4.2
-    [Tags]  Mandelbug
+    # XXX: test is randomly failing under Plone 4.2 only
+    Run keyword if  '${CMFPLONE_VERSION}' >= '4.3'  Remove Tags  Mandelbug
```
Comment (hvelarde) [inline src/collective/cover/tests/test_banner_tile.robot]: @datakurre this seems not to be working; what's the value of `${CMFPLONE_VERSION}` under Plone 4.2?

**Comment by datakurre** [comment]
```
CMFPLONE_VERSION = pkg_resources.get_distribution('Products.CMFPlone').version
if CMFPLONE_VERSION.startswith('4.'):
    CMFPLONE_SELECTORS = 'selectors/cmfplone43.robot'
elif CMFPLONE_VERSION.startswith('5.'):
    CMFPLONE_SELECTORS = 'selectors/cmfplone50.robot'
```

**Comment by datakurre** [comment]
@hvelarde The issue is that you are actually comparing strings ``"4.2.x" >= "4.3"`` which is True.

**Comment by datakurre** [comment]
So, you need two keywords:

```
Run keyword if  '${CMFPLONE_VERSION}'.startswith('4.3')  Remove Tags  Mandelbug
Run keyword if  '${CMFPLONE_VERSION}'.startswith('5')  Remove Tags  Mandelbug
```

I agree that this is not optimal. Probably there's function in pkg_resources, which could be imported into robot test and used to compare version strings.

**Comment by hvelarde** [comment]
@datakurre well, there's something wrong with that:

```python
>>> "4.2.x" >= "4.3"
False
```

**Comment by datakurre** [comment]
@hvelarde Sorry, I was too hasty. So, that should work then. You can use Log keyword to log the variable.

**Comment by hvelarde** [comment]
@datakurre the problem is a different one; as you can see from the report all critical tests are passing:

https://s3-sa-east-1.amazonaws.com/collective-cover/collective/collective.cover/2664/2664.1/parts/test/robot_report.html

but the test command is returning 1:

```
# bin/test -t "Test File Tile"
Running collective.cover.testing.collective.cover:Robot tests:
  Set up plone.testing.zca.LayerCleanup in 0.000 seconds.
  Set up plone.testing.z2.Startup in 0.354 seconds.
  Set up plone.app.testing.layers.PloneFixture in 10.051 seconds.
  Set up collective.cover.testing.Fixture in 10.206 seconds.
  Set up plone.app.robotframework.remote.AutoLoginRemoteLibrary:RobotRemote in 0.009 seconds.
  Set up plone.testing.z2.ZServer in 0.502 seconds.
  Set up collective.cover.testing.collective.cover:Robot in 0.000 seconds.
  Running:
    1/1 (100.0%)
==============================================================================
Test File Tile                                                                
==============================================================================
Test File Tile                                                        | FAIL |
Text 'Edit File Tile' did not disappear in 30 seconds
------------------------------------------------------------------------------
Test File Tile                                                        | PASS |
0 critical tests, 0 passed, 0 failed
1 test total, 0 passed, 1 failed
==============================================================================
Output:  /home/hvelarde/collective/cover/parts/test/test_file_tile/Test_File_Tile/output.xml



Failure in test Test File Tile (test_file_tile.robot)
Traceback (most recent call last):
  File "/home/hvelarde/.buildout/eggs/unittest2-0.5.1-py2.7.egg/unittest2/case.py", line 340, in run
    testMethod()
  File "/home/hvelarde/.buildout/eggs/robotsuite-1.7.0-py2.7.egg/robotsuite/__init__.py", line 461, in runTest
    assert last_status == 'PASS', last_message
AssertionError: Text 'Edit File Tile' did not disappear in 30 seconds

The following test left new threads behind:
Test File Tile (test_file_tile.robot)
New thread(s): [<_DummyThread(Dummy-1, started daemon 139729337575168)>]

  Ran 1 tests with 1 failures and 0 errors in 1 minutes 6.727 seconds.
Tearing down left over layers:
  Tear down collective.cover.testing.collective.cover:Robot in 0.000 seconds.
  Tear down plone.app.robotframework.remote.AutoLoginRemoteLibrary:RobotRemote in 0.005 seconds.
  Tear down plone.testing.z2.ZServer in 5.077 seconds.
  Tear down collective.cover.testing.Fixture in 0.009 seconds.
  Tear down plone.app.testing.layers.PloneFixture in 0.085 seconds.
  Tear down plone.testing.z2.Startup in 0.006 seconds.
  Tear down plone.testing.zca.LayerCleanup in 0.005 seconds.
# echo $?
1
```

this seems to be a bug; where do we need to report it?

**Comment by datakurre** [comment]
There was some reason, why I was not able to automatically detect 
non-critical tests in robotsuite, but those can be defined in test_robot.py:

from https://pypi.python.org/pypi/robotsuite:

Declaring tests non-critical by given set of tags

Robot Framework supports declaring tests with given tags as non-critical 
to prevent their failing to fail the complete build on CI. This is 
supported as keyword argument for RobotTestSuite as follows:

def test_suite():
     suite = unittest.TestSuite()
     suite.addTests([
         layered(RobotTestSuite('mysuite.txt',
                                noncritical=['non-critical-tag']),
                 layer=ACCEPTANCE_TESTING),
     ])
     return suite



Héctor Velarde wrote:

**Comment by hvelarde** [comment]
well, here is working: https://s3-sa-east-1.amazonaws.com/collective-cover/collective/collective.cover/2664/2664.6/parts/test/robot_report.html

**Pull request merged**